### PR TITLE
EOS-19851: All mini-provisioner logs should go to ha_setup.log file

### PR DESCRIPTION
**Pull request by akash2144**
Signed-off-by: Akash Dudhane <akash.a.dudhane@seagate.com>

## Problem Statement
<pre>
  <code>
  Story Ref (if any): https://jts.seagate.com/browse/EOS-19851
    All mini-provisioner logs should go to ha_setup.log file.
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  Yes
  </code>
</pre>
## Problem Description
<pre>
  <code>
    The mini-provisioner logs for all mini-provisioner commands should go to ha_setup.log file always.
  </code>
</pre>
## Solution
<pre>
  <code>
    Previously all the logs related to config command goes to the cluster_manager.log file. Now all the logs are redirected to ha_setup.log file.
  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
    Testing is done to check whether all the logs go to ha_setup file or not. 
  </code>
</pre>

### CODE REVIEW

File: ha/core/cluster/cluster_manager.py [hunk 1]
```diff
@@ -267,12 +267,13 @@ def shutdown(self):
 
 # Note: This class is used by version 2
 class CortxClusterManager:
-    def __init__(self):
+    def __init__(self,logfix = True):
```
Comment (ajay-paratmandali) [inline ha/core/cluster/cluster_manager.py:270]: `logfix` change to `default_log_enable`

File: ha/setup/ha_setup.py [hunk 2]
```diff
@@ -244,7 +244,7 @@ def process(self):
         self._update_env(node_name, node_type, const.HA_CLUSTER_SOFTWARE)
         self._fetch_fids()
         self._update_cluster_manager_config()
-        self._cluster_manager = CortxClusterManager()
+        self._cluster_manager = CortxClusterManager(False)
```
Comment (ajay-paratmandali) [inline ha/setup/ha_setup.py:247]: CortxClusterManager(logfix=False)
if there are more parameter updated in future then above can handle condition

Comment (akash2144) [reply ha/setup/ha_setup.py]: Done.

File: ha/core/cluster/cluster_manager.py [hunk 3]
```diff
@@ -267,12 +267,13 @@ def shutdown(self):
 
 # Note: This class is used by version 2
 class CortxClusterManager:
-    def __init__(self):
+    def __init__(self,default_log_enable = True):
```
Comment (azheregelya) [inline ha/core/cluster/cluster_manager.py:270]: Please format your code according to PEP8 standard. This is also applicable for all other files in this merge request.

**Review by azheregelya — CHANGES_REQUESTED** [review]

**Review by azheregelya — CHANGES_REQUESTED** [review]

File: ha/core/cluster/cluster_manager.py [hunk 4]
```diff
@@ -267,12 +267,13 @@ def shutdown(self):
...
         """
         Manage cluster operation
         """
         # TODO: Update Config manager if log utility changes.
```
[the remark this answers is not in the document]
Comment (akash2144) [reply ha/core/cluster/cluster_manager.py]: yes, I removed that line.

**Review by ajay-paratmandali — APPROVED** [review]

File: ha/core/cluster/cluster_manager.py [hunk 5]
```diff
@@ -267,12 +267,12 @@ def shutdown(self):
 
 # Note: This class is used by version 2
 class CortxClusterManager:
-    def __init__(self):
+    def __init__(self, default_log_enable=True):
```
Comment (indrajitzagade) [inline ha/core/cluster/cluster_manager.py:270]: The argument name is lengthy, but ok.

File: ha/core/cluster/cluster_manager.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ajay-paratmandali) [reply ha/core/cluster/cluster_manager.py]: Yes add todo with EOS-17614 ticket
it is different problem
if log enable in clustermanager then cli log can also go to clouster_manager.log

[the remark this answers is not in the document]
Comment (akash2144) [reply ha/core/cluster/cluster_manager.py]: Added TODO.

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Comment by indrajitzagade** [comment]
Adding test comment so see if the VM deployment works for this PR.

**Comment by gowthamchinna** [comment]
test this please

**Comment by gowthamchinna** [comment]
ok to test

File: ha/core/cluster/cluster_manager.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (indrajitzagade) [reply ha/core/cluster/cluster_manager.py]: Sure

File: ha/core/cluster/cluster_manager.py [hunk 6]
```diff
@@ -267,12 +267,13 @@ def shutdown(self):
...
         """
         Manage cluster operation
         """
-        # TODO: Update Config manager if log utility changes.
-        ConfigManager.init("cluster_manager")
+        # TODO: Update Config manager if log utility changes.(reference EOS-17614)
+        if default_log_enable is True:
```
Comment (indrajitzagade) [inline ha/core/cluster/cluster_manager.py:275]: Sorry, but this is more of a configmanager init to be done or not based on the argument passed. The argument name should be changed accordingly.
cc @ajay-paratmandali

**Review by azheregelya — APPROVED** [review]

File: ha/core/cluster/cluster_manager.py (hunk 6 above)
Comment (akash2144) [reply ha/core/cluster/cluster_manager.py]: Argument name changed to "init_config_manager".

Comment (ajay-paratmandali) [reply ha/core/cluster/cluster_manager.py]: @indrajitzagade  Akash
I think `default_log_enable` is good option 
Here, in recent change we are doing conf init both case only difference is log enabling 
so `init_config_manager` is not hold true here
```
        if init_config_manager is True:
            ConfigManager.init("cluster_manager")
        else:
            ConfigManager.init(None)
```

Comment (indrajitzagade) [reply ha/core/cluster/cluster_manager.py]: @ajay-paratmandali I think I missed this recent change. And yes then the old argument name makes sense. Thanks.

Comment (akash2144) [reply ha/core/cluster/cluster_manager.py:275]: Argument name renamed to "default_log_enable".

**Review by indrajitzagade — APPROVED** [review]

**Pull request merged**